src/messages/manifestjson.js

```diff
@@ -555,3 +555,12 @@ export function permissionFirefoxAndroidUnsupportedByMinVersion(
     file: MANIFEST_JSON,
   };
 }
+
+export const RESTRICTED_HOMEPAGE_URL = {
+  code: 'RESTRICTED_HOMEPAGE_URL',
+  message: i18n._('Linking to AMO is not allowed'),
+  description: i18n._(
+    'Links directing to AMO domain name are not allowed to be used for homepage'
+  ),
```

Comment:
AMO is how we refer to "addons.mozilla.org" internally, but I doubt that it would be immediately understandable from the extension developer that do not already know that, I would replace AMO with the "addons.mozilla.org" (even if technically we would also reject on "addons-dev.allizom.org", but I doubt that will be really a common mistake, it is just a dev instance we use before deploy new AMO versions to the official one).